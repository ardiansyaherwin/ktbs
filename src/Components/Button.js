import React from 'react'
import PropTypes from 'prop-types'

const Button = (props) => {
  const {text, children, style, onClick, onBlur, disabled} = props
  
  return (
    <button 
    	style={style} 
    	onClick={onClick}
    	onBlur={onBlur} 
    	disabled={disabled}
	>
      {typeof children !== 'undefined'
      ? children
      : text
      }
    </button>
  )
}

Button.propTypes = {
  text: PropTypes.string,
  type: PropTypes.string,
  style: PropTypes.object,
  onClick: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  disabled: PropTypes.bool
}

Button.defaultProps = {
  disabled: false,
  onClick: () => {},
}

export default Button
