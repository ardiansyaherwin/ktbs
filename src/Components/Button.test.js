import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

test("Button should accept text, type, style, onClick as props", () => {
	const component = renderer.create(
		<Button onClick={() => null} type={''} text={'TEST'} style={{}}/>
	)
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})