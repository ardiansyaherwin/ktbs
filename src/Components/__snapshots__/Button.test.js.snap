// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should accept text, type, style, onClick as props 1`] = `
<button
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  TEST
</button>
`;
